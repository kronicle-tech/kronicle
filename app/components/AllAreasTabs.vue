<template>
  <div class="my-3">
    <b-button to="/all-areas">Areas</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
