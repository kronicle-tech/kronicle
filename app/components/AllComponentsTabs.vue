<template>
  <div class="my-3">
    <b-button class="my-1" :to="{ name: 'all-components' }">
      Components
    </b-button>
    <b-button
      :to="{ name: 'all-components-tests', query: { testOutcome: 'fail' } }"
    >
      Tests
    </b-button>
    <b-button
      class="my-1"
      :to="{ name: 'all-components-cross-functional-requirements' }"
    >
      Cross Functional Requirements
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-tech-debts' }">
      Tech Debts
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-openapi-specs' }">
      OpenAPI Specs
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-graphql-schemas' }">
      GraphQL Schemas
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-key-software' }">
      Key Software
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-repos' }">
      Repos
    </b-button>
    <b-button class="my-1" :to="{ name: 'all-components-errors' }">
      Errors
    </b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
