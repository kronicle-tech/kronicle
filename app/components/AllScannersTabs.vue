<template>
  <div class="my-3">
    <b-button :to="{ name: 'all-scanners' }">Scanners</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
