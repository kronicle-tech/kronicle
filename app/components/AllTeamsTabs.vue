<template>
  <div class="my-3">
    <b-button to="/all-teams">Teams</b-button>
    <b-button to="/all-teams/components">Components</b-button>
    <b-button to="/all-teams/tech-debts">Tech Debts</b-button>
    <b-button to="/all-teams/tests">Tests</b-button>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BButton } from 'bootstrap-vue'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
})
</script>
