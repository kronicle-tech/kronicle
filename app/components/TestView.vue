<template>
  <div>
    <b-card title="Description" class="my-3">
      <Markdown :markdown="test.description" />
    </b-card>

    <b-card title="Notes" class="my-3">
      <Markdown :markdown="test.notes" />
    </b-card>

    <b-card title="Priority" class="my-3">
      {{ priorityText }}
    </b-card>

    <TestResultsView :test-id="test.id" :components="components" />
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { BCard } from 'bootstrap-vue'
import { Component, Test } from '~/types/kronicle-service'
import { getPriorityText } from '~/src/priorityHelper'
import Markdown from '~/components/Markdown.vue'
import TestResultsView from '~/components/TestResultsView.vue'

export default Vue.extend({
  components: {
    'b-card': BCard,
    Markdown,
    TestResultsView,
  },
  props: {
    test: {
      type: Object as PropType<Test>,
      required: true,
    },
    components: {
      type: Array as PropType<Component[]>,
      required: true,
    },
  },
  computed: {
    priorityText(): string {
      return getPriorityText(this.test.priority)
    },
  },
})
</script>
