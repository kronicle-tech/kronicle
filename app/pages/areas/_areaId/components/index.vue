<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <h1 class="text-info my-3">{{ area.name }} Area - Components</h1>

          <AreaTabs :area-id="area.id" />
        </b-col>
      </b-row>
    </b-container>

    <ComponentsView :components="area.components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BCol, BContainer, BRow} from "bootstrap-vue";
import { Area } from '~/types/kronicle-service'
import AreaTabs from '~/components/AreaTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AreaTabs,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const area = await fetch(
      `${$config.serviceBaseUrl}/v1/areas/${route.params.areaId}?fields=area(id,name,components(id,name,typeId,description,tags,teams,platformId))`
    )
      .then((res) => res.json())
      .then((json) => json.area as Area)

    store.commit('componentFilters/initialize', {
      components: area.components,
      route,
    })

    return {
      area,
    }
  },
  data() {
    return {
      area: {} as Area,
    }
  },
  head(): MetaInfo {
    return {
      title: `Kronicle - ${this.area.name} Area - Components`,
    }
  },
})
</script>
