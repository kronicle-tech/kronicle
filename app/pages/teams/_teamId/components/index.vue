<template>
  <div class="m-3">
    <h1 class="text-info my-3">{{ team.name }} - Components</h1>

    <TeamTabs :team-id="team.id" />

    <ComponentsView :components="team.components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import { Team } from '~/types/kronicle-service'
import ComponentsView from '~/components/ComponentsView.vue'
import TeamTabs from '~/components/TeamTabs.vue'
import { NuxtError } from '~/src/nuxtError'

export default Vue.extend({
  components: {
    ComponentsView,
    TeamTabs,
  },
  async asyncData({ $config, route, store }) {
    const team = await fetch(
      `${$config.serviceBaseUrl}/v1/teams/${route.params.teamId}?fields=team(id,name,components(id,name,discovered,type,description,tags,teams,platformId))`
    )
      .then((res) => res.json())
      .then((json) => json.team as Team | undefined)

    if (!team) {
      throw new NuxtError('Team not found', 404)
    }

    store.commit('componentFilters/initialize', {
      components: team.components,
      route,
    })

    return {
      team,
    }
  },
  data() {
    return {
      team: {} as Team,
    }
  },
  head(): MetaInfo {
    return {
      title: `Kronicle - ${this.team.name} - Components`,
    }
  },
})
</script>
