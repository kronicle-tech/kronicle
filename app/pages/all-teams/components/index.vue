<template>
  <div class="m-3">
    <AllTeamsTabs />

    <b-card-group v-for="(row, rowIndex) in rows" :key="rowIndex" deck>
      <b-card
        v-for="(item, itemIndex) in row"
        :key="itemIndex"
        bg-variant="dark"
        :header="item.team.name"
        text-variant="white"
        class="my-3"
      >
        <b-card
          v-for="component in item.components"
          :key="component.id"
          no-body
          class="mt-1"
        >
          <b-card-header header-tag="header" class="p-1" role="tab">
            <b-button
              v-b-toggle="`accordion-${rowIndex}-${itemIndex}-${component.id}`"
              block
              variant="info"
              >{{ component.name }}</b-button
            >
          </b-card-header>
          <b-collapse
            :id="`accordion-${rowIndex}-${itemIndex}-${component.id}`"
            visible
            :accordion="`accordion-${rowIndex}-${itemIndex}`"
            role="tabpanel"
          >
            <b-card-body>
              <ComponentPanel :component="component" />
            </b-card-body>
          </b-collapse>
        </b-card>
      </b-card>
    </b-card-group>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import {
  BButton,
  BCard,
  BCardBody,
  BCardGroup,
  BCardHeader,
  BCollapse,
  VBToggle,
} from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component, Team } from '~/types/kronicle-service'
import AllTeamsTabs from '~/components/AllTeamsTabs.vue'
import ComponentPanel from '~/components/ComponentPanel.vue'

interface Item {
  team: Team
  components: Component[]
}

export default Vue.extend({
  components: {
    AllTeamsTabs,
    'b-button': BButton,
    'b-card': BCard,
    'b-card-body': BCardBody,
    'b-card-group': BCardGroup,
    'b-card-header': BCardHeader,
    'b-collapse': BCollapse,
    ComponentPanel,
  },
  directives: {
    'b-toggle': VBToggle,
  },
  async asyncData({ $config }) {
    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id,name)`
    )
      .then((res) => res.json())
      .then((json) => json.teams)

    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,teams,tags,description,notes,responsibilities,techDebts,openApiSpecs,links)`
    )
      .then((res) => res.json())
      .then((json) => json.components)

    return {
      teams,
      components,
    }
  },
  data() {
    return {
      teams: [] as Team[],
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Teams',
    }
  },
  computed: {
    rows() {
      const that = this
      const rows = [] as Array<Array<Item>>
      const teamCount = this.teams.length
      let startIndex = 0
      while (startIndex < teamCount - 1) {
        const endIndex = Math.min(startIndex + 3, teamCount)
        const row = that.teams.slice(startIndex, endIndex).map(
          (team) =>
            ({
              team,
              components: getComponents(team),
            } as Item)
        )
        rows.push(row)
        startIndex = endIndex
      }
      return rows

      function getComponents(team: Team) {
        return that.components.filter((component) => {
          return component.teams.some((componentTeam) => {
            return (
              componentTeam.type !== 'previous' &&
              componentTeam.teamId === team.id
            )
          })
        })
      }
    },
  },
})
</script>
