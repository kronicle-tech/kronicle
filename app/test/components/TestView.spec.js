import TestView from '@/components/TestView.vue'
import { createViewComponentWrapper } from '~/test/components/viewUtils'
import { expectTextsInTableRows } from '~/test/components/tableUtils'

describe('TestView', () => {
  let propsData
  let wrapper
  async function createWrapper() {
    wrapper = await createViewComponentWrapper(TestView, {
      propsData,
    })
  }

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when components is set to multiple components with multiple test results', () => {
    beforeEach(() => {
      propsData = {
        test: {
          id: 'test-test-id-1',
          description: 'Test Test Description 1',
          notes: 'Test Test Notes 1',
          priority: 'very-high',
        },
        components: [
          {
            id: 'test-component-id-1',
            name: 'Test Component Name 1',
            testResults: [
              {
                testId: 'test-test-id-1',
                priority: 'very-high',
                outcome: 'fail',
                message: 'Test Message 1a',
              },
              {
                testId: 'test-test-id-2',
                priority: 'high',
                outcome: 'success',
                message: 'Test Message 1b',
              },
            ],
          },
          {
            id: 'test-component-id-2',
            name: 'Test Component Name 2',
            testResults: [
              {
                testId: 'test-test-id-1',
                priority: 'very-high',
                outcome: 'fail',
                message: 'Test Message 2a',
              },
              {
                testId: 'test-test-id-2',
                priority: 'high',
                outcome: 'success',
                message: 'Test Message 2b',
              },
            ],
          },
          {
            id: 'test-component-id-3',
            name: 'Test Component Name 3',
            testResults: [
              {
                testId: 'test-test-id-1',
                priority: 'very-high',
                outcome: 'fail',
                message: 'Test Message 3a',
              },
              {
                testId: 'test-test-id-2',
                priority: 'high',
                outcome: 'success',
                message: 'Test Message 3b',
              },
            ],
          },
        ],
      }
    })

    test('renders the details of the test and the test results of the components and only shows test results for this test', async () => {
      await createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
      expectTextsInTableRows(wrapper, 'td.test-id a', [
        'test-test-id-1',
        'test-test-id-1',
        'test-test-id-1',
      ])
    })
  })
})
