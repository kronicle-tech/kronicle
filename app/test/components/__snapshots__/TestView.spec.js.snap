// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestView when components is set to multiple components with multiple test results renders the details of the test and the test results of the components and only shows test results for this test 1`] = `
"<div>
  <div class=\\"card my-3\\">
    <!---->
    <!---->
    <div class=\\"card-body\\">
      <h4 class=\\"card-title\\">Description</h4>
      <!---->
      <div>
        <p>Test Test Description 1</p>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"card my-3\\">
    <!---->
    <!---->
    <div class=\\"card-body\\">
      <h4 class=\\"card-title\\">Notes</h4>
      <!---->
      <div>
        <p>Test Test Notes 1</p>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"card my-3\\">
    <!---->
    <!---->
    <div class=\\"card-body\\">
      <h4 class=\\"card-title\\">Priority</h4>
      <!---->
      Very High
    </div>
    <!---->
    <!---->
  </div>
  <div>
    <div class=\\"card my-3\\">
      <!---->
      <!---->
      <div class=\\"list-group\\">
        <!---->
        <div class=\\"list-group-item list-group-item-danger fail-count\\"><span class=\\"display-1\\">
          3
        </span> <b>failing</b> tests
        </div>
      </div>
      <!---->
      <!---->
    </div>
    <div class=\\"mt-3 mb-4\\" components=\\"[object Object],[object Object],[object Object]\\"> <button id=\\"toggleFilters\\" type=\\"button\\" class=\\"btn btn-primary collapsed\\" aria-expanded=\\"false\\" aria-controls=\\"filters\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1em\\" height=\\"1em\\" focusable=\\"false\\" role=\\"img\\" aria-label=\\"blank\\" aria-hidden=\\"true\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"currentColor\\" class=\\"bi-blank b-icon bi\\">
          <g></g>
        </svg> Filters
      </button>
      <transition-stub css=\\"true\\" enterclass=\\"\\" leaveclass=\\"collapse show\\" entertoclass=\\"collapse show\\" leavetoclass=\\"collapse\\" enteractiveclass=\\"collapsing\\" leaveactiveclass=\\"collapsing\\" class=\\"mt-3\\">
        <div id=\\"filters\\" class=\\"collapse\\" style=\\"display: none;\\">
          <div class=\\"card-columns\\">
            <!---->
            <!---->
            <div class=\\"card bg-secondary\\">
              <!---->
              <!---->
              <div class=\\"card-body\\">
                <!---->
                <!---->
                <fieldset class=\\"form-group\\" id=\\"__BVID__12\\">
                  <legend tabindex=\\"-1\\" class=\\"bv-no-focus-ring col-form-label pt-0\\" id=\\"__BVID__12__BV_label_\\">Test Outcomes</legend>
                  <div>
                    <div role=\\"group\\" tabindex=\\"-1\\" class=\\"bv-no-focus-ring\\" id=\\"__BVID__13\\">
                      <div class=\\"custom-control custom-checkbox\\"><input type=\\"checkbox\\" name=\\"testOutcome\\" class=\\"custom-control-input\\" value=\\"fail\\" id=\\"__BVID__13_BV_option_0\\"><label class=\\"custom-control-label\\" for=\\"__BVID__13_BV_option_0\\"><span>fail</span></label></div>
                      <div class=\\"custom-control custom-checkbox\\"><input type=\\"checkbox\\" name=\\"testOutcome\\" class=\\"custom-control-input\\" value=\\"success\\" id=\\"__BVID__13_BV_option_1\\"><label class=\\"custom-control-label\\" for=\\"__BVID__13_BV_option_1\\"><span>success</span></label></div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
            <div class=\\"card bg-secondary\\">
              <!---->
              <!---->
              <div class=\\"card-body\\">
                <!---->
                <!---->
                <fieldset class=\\"form-group\\" id=\\"__BVID__16\\">
                  <legend tabindex=\\"-1\\" class=\\"bv-no-focus-ring col-form-label pt-0\\" id=\\"__BVID__16__BV_label_\\">Component Types</legend>
                  <div>
                    <div role=\\"group\\" tabindex=\\"-1\\" class=\\"bv-no-focus-ring\\" id=\\"__BVID__17\\">
                      <div class=\\"custom-control custom-checkbox\\"><input type=\\"checkbox\\" name=\\"componentTypeId\\" class=\\"custom-control-input\\" value=\\"\\" id=\\"__BVID__17_BV_option_0\\"><label class=\\"custom-control-label\\" for=\\"__BVID__17_BV_option_0\\"><span>null</span></label></div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
            <div class=\\"card bg-secondary\\">
              <!---->
              <!---->
              <div class=\\"card-body\\">
                <!---->
                <!---->
                <fieldset class=\\"form-group\\" id=\\"__BVID__19\\">
                  <legend tabindex=\\"-1\\" class=\\"bv-no-focus-ring col-form-label pt-0\\" id=\\"__BVID__19__BV_label_\\">Platforms</legend>
                  <div>
                    <div role=\\"group\\" tabindex=\\"-1\\" class=\\"bv-no-focus-ring\\" id=\\"__BVID__20\\">
                      <div class=\\"custom-control custom-checkbox\\"><input type=\\"checkbox\\" name=\\"platformId\\" class=\\"custom-control-input\\" value=\\"undefined\\" id=\\"__BVID__20_BV_option_0\\"><label class=\\"custom-control-label\\" for=\\"__BVID__20_BV_option_0\\"><span>undefined</span></label></div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
            <div class=\\"card bg-secondary\\">
              <!---->
              <!---->
              <div class=\\"card-body\\">
                <!---->
                <!---->
                <div role=\\"group\\" class=\\"form-group\\" id=\\"__BVID__22\\"><label for=\\"component-filter\\" class=\\"d-block\\" id=\\"__BVID__22__BV_label_\\">Component</label>
                  <div><select id=\\"component-filter\\" class=\\"custom-select custom-select-sm\\">
                      <option value=\\"\\">Please select a component</option>
                      <option value=\\"test-component-id-1\\">test-component-id-1</option>
                      <option value=\\"test-component-id-2\\">test-component-id-2</option>
                      <option value=\\"test-component-id-3\\">test-component-id-3</option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
    <div class=\\"card my-3\\">
      <!---->
      <!---->
      <div class=\\"card-body\\">
        <h4 class=\\"card-title\\">3 Test Results</h4>
        <!---->
        <table class=\\"table table-dark\\">
          <thead>
            <tr>
              <th>Component</th>
              <th>Test</th>
              <th>Teams</th>
              <th>Priority</th>
              <th>Outcome</th>
              <th>Message</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td class=\\"test-id\\"><a href=\\"/tests/test-test-id-1\\">test-test-id-1</a></td>
              <td class=\\"test-component table-primary\\"><a href=\\"/components/test-component-id-1\\">Test Component Name 1</a></td>
              <td>
                <!---->
              </td>
              <td class=\\"test-priority\\"><span class=\\"badge badge-danger\\">
          Very High
        </span></td>
              <td class=\\"test-outcome table-danger\\">
                failing
              </td>
              <td class=\\"test-message\\">
                <div>
                  <p>Test Message 1a</p>
                </div>
              </td>
            </tr>
            <tr>
              <td class=\\"test-id\\"><a href=\\"/tests/test-test-id-1\\">test-test-id-1</a></td>
              <td class=\\"test-component table-primary\\"><a href=\\"/components/test-component-id-2\\">Test Component Name 2</a></td>
              <td>
                <!---->
              </td>
              <td class=\\"test-priority\\"><span class=\\"badge badge-danger\\">
          Very High
        </span></td>
              <td class=\\"test-outcome table-danger\\">
                failing
              </td>
              <td class=\\"test-message\\">
                <div>
                  <p>Test Message 2a</p>
                </div>
              </td>
            </tr>
            <tr>
              <td class=\\"test-id\\"><a href=\\"/tests/test-test-id-1\\">test-test-id-1</a></td>
              <td class=\\"test-component table-primary\\"><a href=\\"/components/test-component-id-3\\">Test Component Name 3</a></td>
              <td>
                <!---->
              </td>
              <td class=\\"test-priority\\"><span class=\\"badge badge-danger\\">
          Very High
        </span></td>
              <td class=\\"test-outcome table-danger\\">
                failing
              </td>
              <td class=\\"test-message\\">
                <div>
                  <p>Test Message 3a</p>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>"
`;
